<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>The tradeSeq workflow • tradeSeq</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="The tradeSeq workflow">
<meta property="og:description" content="tradeSeq">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">tradeSeq</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.3.15</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/tradeSeq.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Monocle.html">Using Monocle as input to **tradeSeq**</a>
    </li>
    <li>
      <a href="../articles/fitGAM.html">Fitting the models and additional control of fitGAM in **tradeSeq**</a>
    </li>
    <li>
      <a href="../articles/multipleConditions.html">Working with multiple conditions</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/statOmics/tradeSeq/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="tradeSeq_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>The tradeSeq workflow</h1>
                        <h4 class="author">Koen Van den Berge and Hector Roux de Bézieux</h4>
            
            <h4 class="date">03/02/2020</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/statOmics/tradeSeq/blob/master/vignettes/tradeSeq.Rmd"><code>vignettes/tradeSeq.Rmd</code></a></small>
      <div class="hidden name"><code>tradeSeq.Rmd</code></div>

    </div>

    
    
<p><code>tradeSeq</code> is an <code>R</code> package that allows analysis of gene expression along trajectories. While it has been developed and applied to single-cell RNA-sequencing (scRNA-seq) data, its applicability extends beyond that, and also allows the analysis of, e.g., single-cell ATAC-seq data along trajectories or bulk RNA-seq time series datasets. For every gene in the dataset, <code>tradeSeq</code> fits a generalized additive model (GAM) by building on the <code>mgcv</code> R package. It then allows statistical inference on the GAM by assessing contrasts of the parameters of the fitted GAM model, aiding in interpreting complex datasets. All details about the <code>tradeSeq</code> model and statistical tests are described in our preprint <span class="citation">(Van den Berge et al. 2019)</span>.</p>
<p>In this vignette, we analyze a subset of the data from <span class="citation">(Paul et al. 2015)</span>. A <code>SingleCellExperiment</code> object of the data has been provided with the <a href="https://github.com/statOmics/tradeSeqpaper"><code>tradeSeq</code></a> package and can be retrieved as shown below. The data and UMAP reduced dimensions were derived from following the <a href="http://cole-trapnell-lab.github.io/monocle-release/monocle3/#tutorial-1-learning-trajectories-with-monocle-3">Monocle 3 vignette</a>.</p>
<p>In this vignette, we use <code>tradeSeq</code> downstream of <code>slinghsot</code><span class="citation">(Street et al. 2018)</span> but it can be used downstream of any trajectory. In particular, if you want to use <code>tradeSeq</code> downstream of <code>monocle</code><span class="citation">(Qiu et al. 2017)</span> and <code>monocle3</code><span class="citation">(Cao et al. 2019)</span>, please refer to our <a href="https://statomics.github.io/tradeSeq/articles/Monocle.html">Monocle vignette</a>.</p>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<p>To install the package, simply run</p>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="co">if</span>(<span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span>(<span class="st">"BiocManager"</span>, quietly = <span class="fl">TRUE</span>)) {
 <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"BiocManager"</span>) 
}
<span class="kw">BiocManager</span>::<span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span>(<span class="st">"tradeSeq"</span>)
</pre></div>
</div>
<div id="load-data" class="section level1">
<h1 class="hasAnchor">
<a href="#load-data" class="anchor"></a>Load data</h1>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="https://statomics.github.io/tradeSeq/index.html">tradeSeq</a></span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw">RColorBrewer</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw">SingleCellExperiment</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw">slingshot</span>)

<span class="co"># For reproducibility</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">RNGversion</a></span>(<span class="st">"3.5.0"</span>)
<span class="fu"><a href="https://rdrr.io/r/grDevices/palette.html">palette</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html">brewer.pal</a></span>(<span class="fl">8</span>, <span class="st">"Dark2"</span>))
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="kw">countMatrix</span>, package = <span class="st">"tradeSeq"</span>)
<span class="kw">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span>(<span class="kw">countMatrix</span>)
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span>(<span class="kw">countMatrix</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="kw">crv</span>, package = <span class="st">"tradeSeq"</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="kw">celltype</span>, package = <span class="st">"tradeSeq"</span>)
</pre></div>
</div>
<div id="fit-negative-binomial-model" class="section level1">
<h1 class="hasAnchor">
<a href="#fit-negative-binomial-model" class="anchor"></a>Fit negative binomial model</h1>
<p>Here we fit the <code>tradeSeq</code> negative binomial generalized additive model (NB-GAM). Please see the <a href="https://statomics.github.io/tradeSeq/articles/fitGAM.html">fitGAM vignette</a> for an extensive description on how to fit the models, tune its options and modify its output.</p>
<p>We first need to decide on the number of knots. This is done using the <strong>evaluateK</strong> function. This takes a little time to run so it is not run here.</p>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="fl">5</span>)
<span class="kw">icMat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/evaluateK.html">evaluateK</a></span>(counts = <span class="kw">counts</span>, sds = <span class="kw">crv</span>, k = <span class="fl">3</span><span class="op">:</span><span class="fl">10</span>, 
                   nGenes = <span class="fl">200</span>, verbose = <span class="kw">T</span>)
</pre></div>
<p><img src="evalK_paul.png"><!-- --></p>
<p>For more explanation on the output from <strong>evaluateK</strong>, we refer users to the <a href="https://statomics.github.io/tradeSeq/articles/fitGAM.html">fitGAM vignette</a>. Here, we pick <code>nknots = 6</code>.</p>
<p>We then fit the models by running the <strong>fitGAM</strong> function. By default, the gene-wise NB-GAM estimates one smoother for every lineage using the negative binomial distribution. Please refer to the <a href="https://statomics.github.io/tradeSeq/articles/fitGAM.html">fitGAM vignette</a> Additional to add additional covariates to the model, speed up computation or allow for custom normalization, amongst others.</p>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="fl">7</span>)
<span class="kw">pseudotime</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/slingshot/man/slingPseudotime.html">slingPseudotime</a></span>(<span class="kw">crv</span>, na = <span class="fl">FALSE</span>)
<span class="kw">cellWeights</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/slingshot/man/slingPseudotime.html">slingCurveWeights</a></span>(<span class="kw">crv</span>)
<span class="kw">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/fitGAM.html">fitGAM</a></span>(counts = <span class="kw">counts</span>, pseudotime = <span class="kw">pseudotime</span>, cellWeights = <span class="kw">cellWeights</span>,
                 nknots = <span class="fl">6</span>, verbose = <span class="fl">FALSE</span>)
</pre></div>
<pre><code>## Warning in any(counts): coercing argument of type 'double' to logical</code></pre>
<p>The model may be hard to fit for some genes, and hence the fitting procedure may not converge for all of the genes in a dataset, especially in datasets with complex patterns and/or many lineages. You can check the convergence of each gene as shown below, where a <code>TRUE</code> value corresponds to a converged model fit, and a <code>FALSE</code> value corresponds to a model that hasn’t been able to converge fully.</p>
<div class="sourceCode" id="cb6"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="fu">rowData</span>(<span class="kw">sce</span>)<span class="op">$</span><span class="kw">tradeSeq</span><span class="op">$</span><span class="kw">converged</span>)
</pre></div>
<pre><code>## 
## TRUE 
##  240</code></pre>
</div>
<div id="within-lineage-comparisons" class="section level1">
<h1 class="hasAnchor">
<a href="#within-lineage-comparisons" class="anchor"></a>Within-lineage comparisons</h1>
<div id="association-of-gene-expression-with-pseudotime" class="section level2">
<h2 class="hasAnchor">
<a href="#association-of-gene-expression-with-pseudotime" class="anchor"></a>Association of gene expression with pseudotime</h2>
<p>A first exploration of the data analysis may consist of checking whether gene expression is associated with a particular lineage. The statistical test performed here, implemented in the <code>associationTest</code> function, is testing the null hypothesis that all smoother coefficients are equal to each other. This can be interpreted as testing whether the average gene expression is significantly changing along pseudotime.</p>
<div class="sourceCode" id="cb8"><pre class="downlit">
<span class="kw">assoRes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/associationTest.html">associationTest</a></span>(<span class="kw">sce</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="kw">assoRes</span>)
</pre></div>
<pre><code>##        waldStat df       pvalue meanLogFC
## Acin1 130.01805  9 0.000000e+00 0.2912046
## Actb  475.85659  9 0.000000e+00 0.2087042
## Ak2    94.12753  9 2.220446e-16 0.2063221
## Alad  194.06872  9 0.000000e+00 0.3610214
## Alas1 977.86093  9 0.000000e+00 0.4469987
## Aldoa 254.86967  9 0.000000e+00 0.1773302</code></pre>
</div>
<div id="discovering-progenitor-marker-genes" class="section level2">
<h2 class="hasAnchor">
<a href="#discovering-progenitor-marker-genes" class="anchor"></a>Discovering progenitor marker genes</h2>
<p>In order to discover marker genes of the progenitor or differentiated cell population, researchers may be interested in assessing differential expression between the progenitor cell population (i.e., the starting point of a lineage) with the differentiated cell type population (i.e., the end point of a lineage). The function <code>startVsEndTest</code> uses a Wald test to assess the null hypothesis that the average expression at the starting point of the smoother (progenitor population) is equal to the average expression at the end point of the smoother (differentiated population). The test basically involves a comparison between two smoother coefficients for every lineage. The function <code>startVsEndTest</code> performs a global test across all lineages by default (i.e. it compares the start and end positions for all lineages simultaneously), but you can also assess all lineages separately by setting <code>lineages=TRUE</code>. Below, we adopt an omnibus test across the two lineages.</p>
<div class="sourceCode" id="cb10"><pre class="downlit">
<span class="kw">startRes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/startVsEndTest.html">startVsEndTest</a></span>(<span class="kw">sce</span>)
</pre></div>
<p>We can visualize the estimated smoothers for the third most significant gene.</p>
<div class="sourceCode" id="cb11"><pre class="downlit">
<span class="kw">oStart</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span>(<span class="kw">startRes</span><span class="op">$</span><span class="kw">waldStat</span>, decreasing = <span class="fl">TRUE</span>)
<span class="kw">sigGeneStart</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="kw">sce</span>)[<span class="kw">oStart</span>[<span class="fl">3</span>]]
<span class="fu"><a href="../reference/plotSmoothers.html">plotSmoothers</a></span>(<span class="kw">sce</span>, <span class="kw">counts</span>, gene = <span class="kw">sigGeneStart</span>)
</pre></div>
<p><img src="tradeSeq_files/figure-html/unnamed-chunk-10-1.png" width="40%"></p>
<p>Alternatively, we can color the cells in UMAP space with that gene’s expression.</p>
<div class="sourceCode" id="cb12"><pre class="downlit">
<span class="fu"><a href="../reference/plotGeneCount.html">plotGeneCount</a></span>(<span class="kw">crv</span>, <span class="kw">counts</span>, gene = <span class="kw">sigGeneStart</span>)
</pre></div>
<p><img src="tradeSeq_files/figure-html/unnamed-chunk-11-1.png" width="50%"></p>
</div>
<div id="comparing-specific-pseudotime-values-within-a-lineage" class="section level2">
<h2 class="hasAnchor">
<a href="#comparing-specific-pseudotime-values-within-a-lineage" class="anchor"></a>Comparing specific pseudotime values within a lineage</h2>
<p>The <code>startVsEndTest</code> compares two points on a lineage, and by default it is comparing the start point with the end point. However, this is a specific form of a more general capability of the <code>startVsEndTest</code> to compare any two points on any lineage. If the interest lies in comparing any two custom pseudotime values, one can specify this using the <code>pseudotimeValues</code> arguments in <code>startVsEndTest</code>. For example, below we’d like to compare the expression for each gene at pseudotime values of <span class="math inline">\(0.8\)</span> and <span class="math inline">\(0.1\)</span>.</p>
<div class="sourceCode" id="cb13"><pre class="downlit">
<span class="kw">customRes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/startVsEndTest.html">startVsEndTest</a></span>(<span class="kw">sce</span>, pseudotimeValues = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0.1</span>, <span class="fl">0.8</span>))
</pre></div>
</div>
</div>
<div id="between-lineage-comparisons" class="section level1">
<h1 class="hasAnchor">
<a href="#between-lineage-comparisons" class="anchor"></a>Between-lineage comparisons</h1>
<div id="discovering-differentiated-cell-type-markers" class="section level2">
<h2 class="hasAnchor">
<a href="#discovering-differentiated-cell-type-markers" class="anchor"></a>Discovering differentiated cell type markers</h2>
<p><code>tradeSeq</code> can discover marker genes for the differentiated cell types by comparing the average expression between end points of the lineage-specific smoothers. This is implemented in the <code>diffEndTest</code> function. By default, <code>diffEndTest</code> performs a global test, testing the null hypothesis that the average expression at the endpoints is equal for all lineages using a multivariate Wald test. If more than two lineages are present, one can assess all pairwise comparisons using the <code>pairwise=TRUE</code> argument.</p>
<div class="sourceCode" id="cb14"><pre class="downlit">
<span class="kw">endRes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/diffEndTest.html">diffEndTest</a></span>(<span class="kw">sce</span>)
</pre></div>
<p>We can plot the most significant gene using the <code>plotSmoothers</code> function.</p>
<div class="sourceCode" id="cb15"><pre class="downlit">
<span class="kw">o</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span>(<span class="kw">endRes</span><span class="op">$</span><span class="kw">waldStat</span>, decreasing = <span class="fl">TRUE</span>)
<span class="kw">sigGene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="kw">sce</span>)[<span class="kw">o</span>[<span class="fl">1</span>]]
<span class="fu"><a href="../reference/plotSmoothers.html">plotSmoothers</a></span>(<span class="kw">sce</span>, <span class="kw">counts</span>, <span class="kw">sigGene</span>)
</pre></div>
<p><img src="tradeSeq_files/figure-html/unnamed-chunk-14-1.png" width="40%"></p>
<p>Alternatively, we can color the cells in UMAP space with that gene’s expression.</p>
<div class="sourceCode" id="cb16"><pre class="downlit">
<span class="fu"><a href="../reference/plotGeneCount.html">plotGeneCount</a></span>(<span class="kw">crv</span>, <span class="kw">counts</span>, gene = <span class="kw">sigGene</span>)
</pre></div>
<p><img src="tradeSeq_files/figure-html/unnamed-chunk-15-1.png" width="50%"></p>
</div>
<div id="discovering-genes-with-different-expression-patterns" class="section level2">
<h2 class="hasAnchor">
<a href="#discovering-genes-with-different-expression-patterns" class="anchor"></a>Discovering genes with different expression patterns</h2>
<p>Asides from testing at the level of the differentiated cell type, researchers may be interested in assessing the expression pattern of a gene over pseudotime. The function <code>patternTest</code> implements a statistical method that checks whether the smoothed gene expression is equal along pseudotime between two or multiple lineages. In practice, we use <span class="math inline">\(100\)</span> points, equally distributed along pseudotime, that are compared between two (or multiple) lineages, and this number can be changed using the <code>nPoints</code> argument.</p>
<div class="sourceCode" id="cb17"><pre class="downlit">
<span class="kw">patternRes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/patternTest.html">patternTest</a></span>(<span class="kw">sce</span>)
<span class="kw">oPat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span>(<span class="kw">patternRes</span><span class="op">$</span><span class="kw">waldStat</span>, decreasing = <span class="fl">TRUE</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">patternRes</span>)[<span class="kw">oPat</span>])
</pre></div>
<pre><code>## [1] "Mpo"   "Prtn3" "Car2"  "Ctsg"  "Elane" "Calr"</code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit">
<span class="fu"><a href="../reference/plotSmoothers.html">plotSmoothers</a></span>(<span class="kw">sce</span>, <span class="kw">counts</span>, gene = <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">patternRes</span>)[<span class="kw">oPat</span>][<span class="fl">4</span>])
</pre></div>
<p><img src="tradeSeq_files/figure-html/unnamed-chunk-16-1.png" width="40%"></p>
<div class="sourceCode" id="cb20"><pre class="downlit">
<span class="fu"><a href="../reference/plotGeneCount.html">plotGeneCount</a></span>(<span class="kw">crv</span>, <span class="kw">counts</span>, gene = <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">patternRes</span>)[<span class="kw">oPat</span>][<span class="fl">4</span>])
</pre></div>
<p><img src="tradeSeq_files/figure-html/unnamed-chunk-17-1.png" width="50%"></p>
<p>The <code>patternTest</code> has an argument <code>eigenThresh</code> that corresponds to an eigenvalue threshold for deciding on the rank of the variance-covariance matrix of the contrasts defined by the <code>patternTest</code>. Lower values are more lenient to adding more information but also decrease computational stability. While the default used to correspond to <code>1e-8</code>, we have recently increased this to <code>1e-2</code> for increased computational stability.</p>
<div id="example-on-combining-patterntest-with-diffendtest-results" class="section level3">
<h3 class="hasAnchor">
<a href="#example-on-combining-patterntest-with-diffendtest-results" class="anchor"></a>Example on combining <code>patternTest</code> with <code>diffEndTest</code> results</h3>
<p>We find genes at the top that are also ranked as DE for the differentiated cell type. What is especially interesting are genes that have different expression patterns but no different expression at the differentiated cell type level. We therefore sort the genes according to the sum of square of their rank in increasing Wald statistics for the <code>patternTest</code> and their rank in decreasing Wald statistics for the <code>diffEndTest</code>.</p>
<div class="sourceCode" id="cb21"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="http://ggplot2.tidyverse.org">ggplot2</a></span>)

<span class="kw">patternRes</span><span class="op">$</span><span class="kw">Gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">patternRes</span>)
<span class="kw">patternRes</span><span class="op">$</span><span class="kw">pattern</span> <span class="op">&lt;-</span> <span class="kw">patternRes</span><span class="op">$</span><span class="kw">waldStat</span>
<span class="kw">patternRes</span> <span class="op">&lt;-</span> <span class="kw">patternRes</span>[, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Gene"</span>, <span class="st">"pattern"</span>)]

<span class="kw">endRes</span><span class="op">$</span><span class="kw">Gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">endRes</span>)
<span class="kw">endRes</span><span class="op">$</span><span class="kw">end</span> <span class="op">&lt;-</span> <span class="kw">endRes</span><span class="op">$</span><span class="kw">waldStat</span>
<span class="kw">endRes</span> <span class="op">&lt;-</span> <span class="kw">endRes</span>[, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Gene"</span>, <span class="st">"end"</span>)]

<span class="kw">compare</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html">merge</a></span>(<span class="kw">patternRes</span>, <span class="kw">endRes</span>, by = <span class="st">"Gene"</span>, all = <span class="fl">FALSE</span>)
<span class="kw">compare</span><span class="op">$</span><span class="kw">transientScore</span> <span class="op">&lt;-</span> 
  <span class="fu"><a href="https://rdrr.io/r/base/rank.html">rank</a></span>(<span class="op">-</span><span class="kw">compare</span><span class="op">$</span><span class="kw">end</span>, ties.method = <span class="st">"min"</span>)<span class="op">^</span><span class="fl">2</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/base/rank.html">rank</a></span>(<span class="kw">compare</span><span class="op">$</span><span class="kw">pattern</span>, ties.method = <span class="st">"random"</span>)<span class="op">^</span><span class="fl">2</span>

<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="kw">compare</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(x = <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log</a></span>(<span class="kw">pattern</span>), y = <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log</a></span>(<span class="kw">end</span>))) <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span>(<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(col = <span class="kw">transientScore</span>)) <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span>(x = <span class="st">"patternTest Wald Statistic (log scale)"</span>,
       y = <span class="st">"diffEndTest Wald Statistic (log scale)"</span>) <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_colour_continuous.html">scale_color_continuous</a></span>(low = <span class="st">"yellow"</span>, high = <span class="st">"red"</span>) <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_classic</a></span>()
</pre></div>
<p><img src="tradeSeq_files/figure-html/unnamed-chunk-18-1.png" width="50%"></p>
<p>Or, we can visualize the expression in UMAP space of the top gene.</p>
<div class="sourceCode" id="cb22"><pre class="downlit">
<span class="kw">topTransient</span> <span class="op">&lt;-</span> <span class="kw">compare</span>[<span class="fu"><a href="https://rdrr.io/r/base/which.min.html">which.max</a></span>(<span class="kw">compare</span><span class="op">$</span><span class="kw">transientScore</span>), <span class="st">"Gene"</span>]
<span class="fu"><a href="../reference/plotSmoothers.html">plotSmoothers</a></span>(<span class="kw">sce</span>, <span class="kw">counts</span>, gene = <span class="kw">topTransient</span>)
</pre></div>
<p><img src="tradeSeq_files/figure-html/unnamed-chunk-19-1.png" width="40%"></p>
<div class="sourceCode" id="cb23"><pre class="downlit">
<span class="fu"><a href="../reference/plotGeneCount.html">plotGeneCount</a></span>(<span class="kw">crv</span>, <span class="kw">counts</span>, gene = <span class="kw">topTransient</span>)
</pre></div>
<p><img src="tradeSeq_files/figure-html/unnamed-chunk-20-1.png" width="50%"></p>
<p>Interestingly, we recover the Irf8 gene in the top 5 genes according to that ranking.</p>
<div class="sourceCode" id="cb24"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(
  <span class="kw">compare</span>[<span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span>(<span class="kw">compare</span><span class="op">$</span><span class="kw">transientScore</span>, decreasing = <span class="fl">TRUE</span>), <span class="st">"Gene"</span>],
  n = <span class="fl">5</span>
)
</pre></div>
<pre><code>## [1] "Mt1"    "Irf8"   "Nedd4"  "Hint1"  "Eif4g1"</code></pre>
<p>We can also plot the Irf8 gene.</p>
<div class="sourceCode" id="cb26"><pre class="downlit">
<span class="fu"><a href="../reference/plotSmoothers.html">plotSmoothers</a></span>(<span class="kw">sce</span>, <span class="kw">counts</span>, gene = <span class="st">"Irf8"</span>)
</pre></div>
<p><img src="tradeSeq_files/figure-html/unnamed-chunk-22-1.png" width="40%"></p>
<div class="sourceCode" id="cb27"><pre class="downlit">
<span class="fu"><a href="../reference/plotGeneCount.html">plotGeneCount</a></span>(<span class="kw">crv</span>, <span class="kw">counts</span>, gene = <span class="st">"Irf8"</span>)
</pre></div>
<p><img src="tradeSeq_files/figure-html/unnamed-chunk-23-1.png" width="50%"></p>
</div>
</div>
<div id="early-drivers-of-differentiation" class="section level2">
<h2 class="hasAnchor">
<a href="#early-drivers-of-differentiation" class="anchor"></a>Early drivers of differentiation</h2>
<p>Another question of interest is to find a list of genes that are differentially expressed between lineages at a particular region, e.g. around the separation of two or multiple lineages. The function <code>earlyDETest</code> implements a statistical method to test the null hypothesis of whether the average gene expression smoothers are equal between lineages in a region defined by two user-specified knots. Again, the knots can be visualized with the <code>plotGeneCount</code> function. By selecting the region covering the first two knot points to test for differential patterns between the lineages, we check which genes are behaving differently around the bifurcation point.</p>
<div class="sourceCode" id="cb28"><pre class="downlit">
<span class="fu"><a href="../reference/plotGeneCount.html">plotGeneCount</a></span>(curve = <span class="kw">crv</span>, counts = <span class="kw">counts</span>,
              clusters = <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/slingshot/man/slingClusterLabels.html">slingClusterLabels</a></span>(<span class="kw">crv</span>), <span class="fl">1</span>, <span class="kw">which.max</span>),
              models = <span class="kw">sce</span>)
</pre></div>
<p><img src="tradeSeq_files/figure-html/unnamed-chunk-24-1.png" width="50%"></p>
<div class="sourceCode" id="cb29"><pre class="downlit">
<span class="kw">earlyDERes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/earlyDETest.html">earlyDETest</a></span>(<span class="kw">sce</span>, knots = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>, <span class="fl">2</span>))
<span class="kw">oEarly</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span>(<span class="kw">earlyDERes</span><span class="op">$</span><span class="kw">waldStat</span>, decreasing = <span class="fl">TRUE</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">earlyDERes</span>)[<span class="kw">oEarly</span>])
</pre></div>
<pre><code>## [1] "Car1"  "Car2"  "Klf1"  "Vim"   "Mt1"   "Ermap"</code></pre>
<div class="sourceCode" id="cb31"><pre class="downlit">
<span class="fu"><a href="../reference/plotSmoothers.html">plotSmoothers</a></span>(<span class="kw">sce</span>, <span class="kw">counts</span>, gene = <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">earlyDERes</span>)[<span class="kw">oEarly</span>][<span class="fl">2</span>])
</pre></div>
<p><img src="tradeSeq_files/figure-html/unnamed-chunk-25-1.png" width="40%"></p>
<div class="sourceCode" id="cb32"><pre class="downlit">
<span class="fu"><a href="../reference/plotGeneCount.html">plotGeneCount</a></span>(<span class="kw">crv</span>, <span class="kw">counts</span>, gene = <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">earlyDERes</span>)[<span class="kw">oEarly</span>][<span class="fl">2</span>])
</pre></div>
<p><img src="tradeSeq_files/figure-html/unnamed-chunk-26-1.png" width="50%"></p>
</div>
</div>
<div id="differential-expression-in-large-datasets" class="section level1">
<h1 class="hasAnchor">
<a href="#differential-expression-in-large-datasets" class="anchor"></a>Differential expression in large datasets</h1>
<p>In large datasets with thousands of single cells, small fold changes can become statistically significant due to the very high sample size. However, these differences may not always be biologically meaningful. To tackle this problem, all DE tests in <code>tradeSeq</code> have an <code>l2fc</code> argument which specifies the absolute value of the log2 fold-change cut-off to test against. For example, setting <code>l2fc=log2(2)</code> will test which genes have a fold change that is significantly higher than 2 or significantly lower than 1/2 within or between the lineages of interest.</p>
<div class="sourceCode" id="cb33"><pre class="downlit">
<span class="co"># testing against fold change threshold of 2</span>
<span class="kw">start2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/startVsEndTest.html">startVsEndTest</a></span>(<span class="kw">sce</span>, l2fc = <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log2</a></span>(<span class="fl">2</span>))
<span class="co"># testing against fold change threshold of 1.5</span>
<span class="kw">pat2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/patternTest.html">patternTest</a></span>(<span class="kw">sce</span>, l2fc = <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log2</a></span>(<span class="fl">1.5</span>))
</pre></div>
</div>
<div id="clustering-of-genes-according-to-their-expression-pattern" class="section level1">
<h1 class="hasAnchor">
<a href="#clustering-of-genes-according-to-their-expression-pattern" class="anchor"></a>Clustering of genes according to their expression pattern</h1>
<div id="extracting-fitted-values-to-use-with-any-clustering-method" class="section level2">
<h2 class="hasAnchor">
<a href="#extracting-fitted-values-to-use-with-any-clustering-method" class="anchor"></a>Extracting fitted values to use with any clustering method</h2>
<p>While <code>tradeSeq</code> provides readily implemented functionality to cluster genes using the <code>clusterExperiment</code> package, users may want to cluster genes using other clustering methods. Fitted values from the <code>tradeSeq</code> models, which can subsequently be used for clustering, can be extracted using the <code>predictSmooth</code> or <code>predictCells</code> functions. <code>predictCells</code> predicts the estimated expression for each cell (i.e. it is a fitted value for that cell) on the count scale. <code>predictSmooth</code>, instead, returns values of the estimated smoother on a grid of pseudotimes. Either of these may be used as input to a clustering method; roughly, <code>predictSmooth</code> may be considered as a cleaner version of <code>predictCells</code>. By default <code>predictSmooth</code> returns a <code>data.frame</code> specifying the estimated smoother for each gene. A matrix containing these values, where each row is a gene, and each column is a point along the lineage on the uniform grid, can be obtained by setting <code>tidy=FALSE</code> in <code>predictSmooth</code>.</p>
<div class="sourceCode" id="cb34"><pre class="downlit">
<span class="kw">yhat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/predictCells.html">predictCells</a></span>(models = <span class="kw">sce</span>, gene = <span class="st">"Irf8"</span>)

<span class="kw">ysmooth</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/predictSmooth.html">predictSmooth</a></span>(models = <span class="kw">sce</span>, gene = <span class="st">"Irf8"</span>, nPoints = <span class="fl">40</span>)
</pre></div>
</div>
<div id="clustering-using-rsec-clusterexperiment" class="section level2">
<h2 class="hasAnchor">
<a href="#clustering-using-rsec-clusterexperiment" class="anchor"></a>Clustering using RSEC, clusterExperiment</h2>
<p>tradeSeq provides the functionality to cluster genes according to their expression pattern along the lineages with the <code>clusterExpressionPatterns</code> function. A number of equally spaced points for every lineage are selected to perform the clustering, and the number of points can be selected with the <code>nPoints</code> argument. The <code>genes</code> argument specifies which genes you want to cluster (e.g., all genes with differential expression patterns). Here, we use 20 points along each lineage to cluster the first 40 genes in the dataset. The clustering itself occurs by the <code>clusterExperiment</code> package <span class="citation">(Risso et al. 2018)</span>, hence the user may select any clustering algorithm that’s built into that package, or custom clustering algorithms implemented by the user. For a list of built-in clustering algorithms within <code>clusterExperiment</code>, run <code><a href="https://rdrr.io/pkg/clusterExperiment/man/builtInClusteringFunctions.html">clusterExperiment::listBuiltInFunctions()</a></code> on the command line.</p>
<div class="sourceCode" id="cb35"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw">clusterExperiment</span>)
<span class="kw">nPointsClus</span> <span class="op">&lt;-</span> <span class="fl">20</span>
<span class="kw">clusPat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/clusterExpressionPatterns.html">clusterExpressionPatterns</a></span>(<span class="kw">sce</span>, nPoints = <span class="kw">nPointsClus</span>,
                                     genes = <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">counts</span>)[<span class="fl">1</span><span class="op">:</span><span class="fl">100</span>])
</pre></div>
<pre><code>## 36 parameter combinations, 36 use sequential method, 36 use subsampling method
## Running Clustering on Parameter Combinations...
## done.</code></pre>
<div class="sourceCode" id="cb37"><pre class="downlit">
<span class="kw">clusterLabels</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterExperiment/man/ClusterExperiment-methods.html">primaryCluster</a></span>(<span class="kw">clusPat</span><span class="op">$</span><span class="kw">rsec</span>)
</pre></div>
<p>The first 4 clusters can be visualized using the normalized expression upon which the clustering is based. Please note that the code below would only works for a trajectory with two lineages. Modify the code appropriately if using with a dataset with 3 lineages or more.</p>
<div class="sourceCode" id="cb38"><pre class="downlit">
<span class="kw">cUniq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(<span class="kw">clusterLabels</span>)
<span class="kw">cUniq</span> <span class="op">&lt;-</span> <span class="kw">cUniq</span>[<span class="op">!</span><span class="kw">cUniq</span> <span class="op">==</span> <span class="op">-</span><span class="fl">1</span>] <span class="co"># remove unclustered genes</span>

<span class="co">for</span> (<span class="kw">xx</span> <span class="co">in</span> <span class="kw">cUniq</span>[<span class="fl">1</span><span class="op">:</span><span class="fl">4</span>]) {
  <span class="kw">cId</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="kw">clusterLabels</span> <span class="op">==</span> <span class="kw">xx</span>)
  <span class="kw">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(data = <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(x = <span class="fl">1</span><span class="op">:</span><span class="kw">nPointsClus</span>,
                                y = <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/range.html">range</a></span>(<span class="kw">clusPat</span><span class="op">$</span><span class="kw">yhatScaled</span>[<span class="kw">cId</span>, ]),
                                        <span class="kw">nPointsClus</span> <span class="op">/</span> <span class="fl">2</span>)),
              <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(x = <span class="kw">x</span>, y = <span class="kw">y</span>)) <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span>(alpha = <span class="fl">0</span>) <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span>(title = <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="st">"Cluster "</span>, <span class="kw">xx</span>),  x = <span class="st">"Pseudotime"</span>, y = <span class="st">"Normalized expression"</span>) <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_classic</a></span>() <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(plot.title = <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span>(hjust = <span class="fl">0.5</span>))
  <span class="co">for</span> (<span class="kw">ii</span> <span class="co">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="kw">cId</span>)) {
    <span class="kw">geneId</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw">clusPat</span><span class="op">$</span><span class="kw">yhatScaled</span>)[<span class="kw">cId</span>[<span class="kw">ii</span>]]
    <span class="kw">p</span> <span class="op">&lt;-</span> <span class="kw">p</span> <span class="op">+</span>
      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html">geom_line</a></span>(data = <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(x = <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="fl">1</span><span class="op">:</span><span class="kw">nPointsClus</span>, <span class="fl">2</span>),
                                  y = <span class="kw">clusPat</span><span class="op">$</span><span class="kw">yhatScaled</span>[<span class="kw">geneId</span>, ],
                                  lineage = <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="fl">0</span><span class="op">:</span><span class="fl">1</span>, each = <span class="kw">nPointsClus</span>)),
                <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(col = <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(<span class="kw">lineage</span>), group = <span class="kw">lineage</span>), lwd = <span class="fl">1.5</span>)
  }
  <span class="kw">p</span> <span class="op">&lt;-</span> <span class="kw">p</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html">guides</a></span>(color = <span class="fl">FALSE</span>) <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_color_manual</a></span>(values = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"orange"</span>, <span class="st">"darkseagreen3"</span>),
                       breaks = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"0"</span>, <span class="st">"1"</span>))  
  <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw">p</span>)
}
</pre></div>
<p><img src="tradeSeq_files/figure-html/plot%20clustering-1.png" width="25%"><img src="tradeSeq_files/figure-html/plot%20clustering-2.png" width="25%"><img src="tradeSeq_files/figure-html/plot%20clustering-3.png" width="25%"><img src="tradeSeq_files/figure-html/plot%20clustering-4.png" width="25%"></p>
</div>
</div>
<div id="contributing-and-requesting" class="section level1">
<h1 class="hasAnchor">
<a href="#contributing-and-requesting" class="anchor"></a>Contributing and requesting</h1>
<p>A number of tests have been implemented in tradeSeq, but researchers may be interested in other hypotheses that current implementations may not be able to address. We therefore welcome contributions on GitHub on novel tests based on the tradeSeq model. Similarly, you may also request novel tests to be implemented in tradeSeq by the developers, preferably by adding <a href="https://github.com/statOmics/tradeSeq/issues">an issue on the GitHub repository</a>. If we feel that the suggested test is widely applicable, we will implement it in tradeSeq.</p>
</div>
<div id="session" class="section level1">
<h1 class="hasAnchor">
<a href="#session" class="anchor"></a>Session</h1>
<div class="sourceCode" id="cb39"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()
</pre></div>
<pre><code>## R version 4.0.2 (2020-06-22)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Catalina 10.15.5
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib
## 
## Random number generation:
##  RNG:     Mersenne-Twister 
##  Normal:  Inversion 
##  Sample:  Rounding 
##  
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] clusterExperiment_2.9.0     ggplot2_3.3.2              
##  [3] slingshot_1.7.2             princurve_2.1.4            
##  [5] SingleCellExperiment_1.11.6 SummarizedExperiment_1.19.6
##  [7] DelayedArray_0.15.6         matrixStats_0.56.0         
##  [9] Matrix_1.2-18               Biobase_2.49.0             
## [11] GenomicRanges_1.41.5        GenomeInfoDb_1.25.8        
## [13] IRanges_2.23.10             S4Vectors_0.27.12          
## [15] BiocGenerics_0.35.4         RColorBrewer_1.1-2         
## [17] tradeSeq_1.3.15             knitr_1.29                 
## 
## loaded via a namespace (and not attached):
##   [1] uuid_0.1-4             backports_1.1.8        VGAM_1.1-3            
##   [4] NMF_0.22.0             plyr_1.8.6             igraph_1.2.5          
##   [7] lazyeval_0.2.2         splines_4.0.2          BiocParallel_1.23.2   
##  [10] densityClust_0.3       rncl_0.8.4             fastICA_1.2-2         
##  [13] gridBase_0.4-7         digest_0.6.25          foreach_1.5.0         
##  [16] htmltools_0.5.0        viridis_0.5.1          magrittr_1.5          
##  [19] memoise_1.1.0          cluster_2.1.0          doParallel_1.0.15     
##  [22] limma_3.45.7           annotate_1.67.0        docopt_0.7.1          
##  [25] pkgdown_1.5.1.9000     prettyunits_1.1.1      colorspace_1.4-1      
##  [28] blob_1.2.1             ggrepel_0.8.2          xfun_0.15             
##  [31] dplyr_1.0.0            sparsesvd_0.2          crayon_1.3.4          
##  [34] RCurl_1.98-1.2         genefilter_1.71.0      phylobase_0.8.10      
##  [37] survival_3.1-12        iterators_1.0.12       ape_5.4               
##  [40] glue_1.4.1             registry_0.5-1         gtable_0.3.0          
##  [43] zlibbioc_1.35.0        XVector_0.29.3         kernlab_0.9-29        
##  [46] Rhdf5lib_1.11.3        HDF5Array_1.17.3       scales_1.1.1          
##  [49] pheatmap_1.0.12        DBI_1.1.0              edgeR_3.31.4          
##  [52] rngtools_1.5           bibtex_0.4.2.2         Rcpp_1.0.5            
##  [55] viridisLite_0.3.0      xtable_1.8-4           progress_1.2.2        
##  [58] bit_1.1-15.2           httr_1.4.1             FNN_1.1.3             
##  [61] ellipsis_0.3.1         pkgconfig_2.0.3        XML_3.99-0.4          
##  [64] farver_2.0.3           locfit_1.5-9.4         softImpute_1.4        
##  [67] howmany_0.3-1          tidyselect_1.1.0       labeling_0.3          
##  [70] rlang_0.4.7            reshape2_1.4.4         AnnotationDbi_1.51.1  
##  [73] munsell_0.5.0          tools_4.0.2            generics_0.0.2        
##  [76] RSQLite_2.2.0          ade4_1.7-15            evaluate_0.14         
##  [79] stringr_1.4.0          yaml_2.2.1             bit64_0.9-7           
##  [82] fs_1.4.2               DDRTree_0.1.5          purrr_0.3.4           
##  [85] RANN_2.6.1             pbapply_1.4-2          nlme_3.1-148          
##  [88] monocle_2.17.0         slam_0.1-47            xml2_1.3.2            
##  [91] compiler_4.0.2         tibble_3.0.3           RNeXML_2.4.5          
##  [94] stringi_1.4.6          desc_1.2.0             RSpectra_0.16-0       
##  [97] lattice_0.20-41        HSMMSingleCell_1.9.0   vctrs_0.3.1           
## [100] rhdf5filters_1.1.2     pillar_1.4.6           lifecycle_0.2.0       
## [103] combinat_0.0-8         zinbwave_1.11.4        bitops_1.0-6          
## [106] irlba_2.3.3            R6_2.4.1               gridExtra_2.3         
## [109] codetools_0.2-16       MASS_7.3-51.6          assertthat_0.2.1      
## [112] rhdf5_2.33.7           pkgmaker_0.31.1        rprojroot_1.3-2       
## [115] withr_2.2.0            qlcMatrix_0.9.7        GenomeInfoDbData_1.2.3
## [118] locfdr_1.1-8           mgcv_1.8-31            hms_0.5.3             
## [121] grid_4.0.2             tidyr_1.1.0            rmarkdown_2.3.2       
## [124] Rtsne_0.15</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-Cao2019">
<p>Cao, Junyue, Malte Spielmann, Xiaojie Qiu, Xingfan Huang, Daniel M. Ibrahim, Andrew J. Hill, Fan Zhang, et al. 2019. “The Dynamics and Regulators of Cell Fate Decisions Are Revealed by Pseudo-Temporal Ordering of Single Cells.” <em>Nature</em>.</p>
</div>
<div id="ref-Paul2015">
<p>Paul, Franziska, Ya’ara Arkin, Amir Giladi, DiegoÂ Adhemar Jaitin, Ephraim Kenigsberg, Hadas Keren-Shaul, Deborah Winter, et al. 2015. “Transcriptional Heterogeneity and Lineage Commitment in Myeloid Progenitors.” <em>Cell</em> 163 (7): 1663–77. <a href="https://doi.org/10.1016/J.CELL.2015.11.013">https://doi.org/10.1016/J.CELL.2015.11.013</a>.</p>
</div>
<div id="ref-Qiu2017">
<p>Qiu, Xiaojie, Qi Mao, Ying Tang, Li Wang, Raghav Chawla, Hannah A Pliner, and Cole Trapnell. 2017. “Reversed graph embedding resolves complex single-cell trajectories.” <em>Nature Methods</em> 14 (10): 979–82. <a href="https://doi.org/10.1038/nmeth.4402">https://doi.org/10.1038/nmeth.4402</a>.</p>
</div>
<div id="ref-Risso2018">
<p>Risso, Davide, Liam Purvis, Russell B. Fletcher, Diya Das, John Ngai, Sandrine Dudoit, and Elizabeth Purdom. 2018. “clusterExperiment and RSEC: A Bioconductor package and framework for clustering of single-cell and other large gene expression datasets.” Edited by Aaron E. Darling. <em>PLOS Computational Biology</em> 14 (9): e1006378. <a href="https://doi.org/10.1371/journal.pcbi.1006378">https://doi.org/10.1371/journal.pcbi.1006378</a>.</p>
</div>
<div id="ref-Street2018a">
<p>Street, Kelly, Davide Risso, Russell B. Fletcher, Diya Das, John Ngai, Nir Yosef, Elizabeth Purdom, and Sandrine Dudoit. 2018. “Slingshot: cell lineage and pseudotime inference for single-cell transcriptomics.” <em>BMC Genomics</em> 19 (1): 477. <a href="https://doi.org/10.1186/s12864-018-4772-0">https://doi.org/10.1186/s12864-018-4772-0</a>.</p>
</div>
<div id="ref-VandenBerge2019a">
<p>Van den Berge, Koen, Hector Roux de Bézieux, Kelly Street, Wouter Saelens, Robrecht Cannoodt, Yvan Saeys, Sandrine Dudoit, and Lieven Clement. 2019. “Trajectory-based differential expression analysis for single-cell sequencing data.” <em>bioRxiv</em>, May, 623397. <a href="https://doi.org/10.1101/623397">https://doi.org/10.1101/623397</a>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Koen Van den Berge, Hector Roux de Bezieux.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
